### PR TITLE
Calendar Feature

### PR DESCRIPTION
**Pull request by Hobbitsloth**
finished for this sprint

### CODE REVIEW

File: plugins/calendar/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (Legion2) [inline plugins/calendar/.gitignore]: remove this gitignore file

File: plugins/calendar/.project [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>amy-plugin-example</name>
```
Comment (Legion2) [inline plugins/calendar/.project]: change the name

File: plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogicTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	private TestFramework framework;
+
+	private CalendarLogic callog;
+
+	/**
+	 * Initializes the class variables before each test
+	 */
+	@BeforeEach
+	public void setup() {
+		this.callog = this.framework.mockService(CalendarLogic.class);
```
Comment (Legion2) [inline plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogicTest.java]: don't mock the service that should be tested, use `setServiceUnderTest`

File: plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+	 * @param event
+	 *            data of the event
+	 * @param withDate
+	 *            determines if event output is with or without the date
+	 */
+	public void eventCalc(DateTime start, Event event, boolean withDate) {
+		String eventData;
+		String datePart = "";
+		if (start != null) {
+			String startData = start.toString();
```
Comment (Legion2) [inline plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java]: `DateTime` is in ISO-8601 / RFC 3339 format. so use `LocalDateTime.parse()` or other implementations of the specifications and don't write your own parser.

File: plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+			DateTime now = new DateTime(System.currentTimeMillis());
+			Events events = this.calendarService.getService().events().list("primary").setTimeMin(now).setOrderBy("startTime")
+					.setSingleEvents(true).execute();
+			List<Event> items = events.getItems();
+			if (items.isEmpty()) {
+				return "No upcoming events found.";
+			}
+			if(!day) {
+				Date today = new Date();
+				LocalDateTime plusOne = today.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
```
Comment (Legion2) [inline plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java]: why?

Comment (Hobbitsloth) [reply plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java]: to differentiate between today and tomorrow, this was the easiest implementation to get the date for the next day I could find. Maybe I should change the name of the variable day or what do you mean?

File: plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java [hunk 6]
```diff
@@ -139,96 +130,101 @@ public String getEventsByDay(boolean day) {
...
-			eventData = event.getSummary() + datePart  + " at "
-					+ startTime.substring(0, 5) + " until " + endtime.toString().substring(11, 16) + "\n";
-			this.eventList.add(eventData);
+	public void checkDay(LocalDateTime dayToCheck, Event event, boolean withDate) {
+		LocalDateTime startDateTime, endDateTime;
+		LocalDate startDate, checkDate, endDate;
+		boolean allDay;
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+		if (event.getStart().getDate() != null) {
+			startDateTime = LocalDateTime.parse(event.getStart().getDate().toString() + "T00:00:00.000+02:00",
```
Comment (Legion2) [inline plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java]: don't hardcode the timezone

File: .gitignore [hunk 7]
```diff
@@ -18,6 +20,9 @@ dependency-reduced-pom.xml
 .idea
 *.iml
 
+#Other -PatrickGe
+config/*.json
```
Comment (Legion2) [inline .gitignore]: this is not needed anymore

File: Amy_Core_Main.launch [hunk 8]
```diff
@@ -11,6 +11,5 @@
 <stringAttribute key="org.eclipse.jdt.launching.CLASSPATH_PROVIDER" value="org.eclipse.m2e.launchconfig.classpathProvider"/>
 <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="de.unistuttgart.iaas.amyassist.amy.core.Main"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="amy-core"/>
-<stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.m2e.launchconfig.sourcepathProvider"/>
```
Comment (Legion2) [inline Amy_Core_Main.launch]: why is this line removed?

Comment (Hobbitsloth) [reply Amy_Core_Main.launch]: i have no idea, I will put it back
